### PR TITLE
Add Ruby v3 compatibility

### PR DESCRIPTION
**Pull request by archanaserver**

### CODE REVIEW

**Comment by archanaserver** [comment]
@ekohl After all the changes jobs are working fine on GitHub Action, but Jenkins started showing me this error again, though i can't see this on my system. Can you help me on this one?

<details>
<summary>Issue</summary>
Finished in 12.443687s, 5.8664 runs/s, 13.7419 assertions/s.



  1) Failure:

Proxy::Dynflow::ProcessManager::#process#test_0001_accepts a timeout [/home/jenkins/workspace/smart_proxy_dynflow-pr-test/test/process_manager_test.rb:112]:

Expected false to be truthy.



73 runs, 171 assertions, 1 failures, 0 errors, 0 skips

rake aborted!

Command failed with status (1)

/home/jenkins/workspace/smart_proxy_dynflow-pr-test/Rakefile:23:in `block in <top (required)>'

/usr/local/rvm/gems/ruby-3.1.2@smart_proxy_dynflow-pr-test-61-1/bin/ruby_executable_hooks:22:in `eval'

/usr/local/rvm/gems/ruby-3.1.2@smart_proxy_dynflow-pr-test-61-1/bin/ruby_executable_hooks:22:in `<main>'

Tasks: TOP => test:core

(See full trace by running task with --trace)

script returned exit code 1
</details>

File: Rakefile [hunk 1]
```diff
@@ -22,6 +23,22 @@ task :test do
   Rake::Task['test:core'].invoke
 end
 
+begin
+  require 'minitest/reporters'
```
Comment (ehelms) [inline Rakefile:27]: Can you extract this change out to it's own dedicated PR? I think that would help identify some of the issues and focus the PRs.

Comment (archanaserver) [reply Rakefile]: #122 it has dependencies with other changes from this PR though it is failing.

**Review by ekohl — COMMENTED** [review]
@adamruzicka I think this looks pretty good, but could you check the RuboCop preferences?

**Comment by adamruzicka** [comment]
If I were doing this, I'd probably keep Layout/EmptyLineAfterMagicComment just to keep things neat, but it really boils down to personal perference. I don't mind having things the way they are right now.

Also this now needs a rebase

**Comment by archanaserver** [comment]
> If I were doing this, I'd probably keep Layout/EmptyLineAfterMagicComment just to keep things neat, but it really boils down to personal perference. I don't mind having things the way they are right now.

Sure, I will raise a new PR to update Layout/EmptyLineAfterMagicComment changes. 
> 
> Also this now needs a rebase

But about the rebase, these changes has already been merged from this PR https://github.com/theforeman/smart_proxy_dynflow/pull/122. To separate Ruby, Rubocop & Jenkins changes, I dropped this one earlier and raised all separately. I guess, I should probably add this one into draft earlier.

**Comment by archanaserver** [comment]
@adamruzicka Layout/EmptyLineAfterMagicComment changes: https://github.com/theforeman/smart_proxy_dynflow/pull/129

**Pull request closed without merging**